src/sphinxcontrib/towncrier/__init__.py

```diff
@@ -71,11 +71,26 @@ def _get_changelog_draft_entries(
         # https://github.com/twisted/towncrier/pull/157#issuecomment-666549246
         # https://github.com/twisted/towncrier/issues/269
         extra_cli_args += '--config', str(config_path)
-    towncrier_output = subprocess.check_output(  # noqa: S603
-        TOWNCRIER_DRAFT_CMD + extra_cli_args,
-        cwd=str(working_dir) if working_dir else None,
-        universal_newlines=True,  # this arg has "text" alias since Python 3.7
-    ).strip()
+
+    try:
+        towncrier_output = subprocess.check_output(  # noqa: S603
+            TOWNCRIER_DRAFT_CMD + extra_cli_args,
+            cwd=str(working_dir) if working_dir else None,
+            stderr=subprocess.PIPE,
+            universal_newlines=True,  # a "text" alias exists since Python 3.7
+        ).strip()
+
+    except subprocess.CalledProcessError as proc_exc:
+        cmd = ' '.join(proc_exc.cmd)
+        stdout = proc_exc.stdout or '[No output]'
+        stderr = proc_exc.stderr or '[No output]'
+        raise RuntimeError(
+            'Command exited unexpectedly\n\n'
+            f'Command: {cmd}\n'
+            f'Return code: {proc_exc.returncode}\n\n'
+            f'Standard output:\n{stdout}\n\n'
+            f'Standard error:\n{stderr}',
+        )
```

Comment:
Let's set the cause explicitly
```suggestion
        ) from proc_exc
```